### ISSUE TITLE
[enhancement] custom auto correct (same as --> undated into →)

### ISSUE THREAD

**Issue by gandrille** [zim-desktop-wiki/zim-desktop-wiki#846]
When typing "-->" is automatically updated into "→". This is great :-)

I would love having a dialog to add/remove other replacements, such as:

coeur into cœur (have you noticed the single unicode character for œ? It's really painful to write this kind of word in french!)
!= into ≠
<< into «
>> into »

Since the replacements are user dependent, it could be nice to have the ability to edit them.

Note: this feature exists in Word (and I have used it intensively in the past!) For example, have a look at https://www.groovypost.com/howto/howto/word-2010-autocorrect-adds-symbols-and-replaces-words/
![image209](https://user-images.githubusercontent.com/621343/57879690-5069b380-781d-11e9-8394-6db574cb7092.png)

**Comment by sojusnik** [comment]
Never noticed that Zim has an auto correct feature, as depicted with -->. Are there other examples as well? Is this somewhere documented?

Would love to have an ability to add my own abbreviations.

For other examples you could have a look on LibreOffice Writer.

**Comment by jaap-karssenberg** [comment]
This replacement is done by the "insert symbol" plugin. Some of your
examples can be put in the config file of that plugin (see the edit button
on the insert symbol dialog).

Op do 16 mei 2019 21:18 schreef sojusnik <notifications@github.com>:

**Comment by gandrille** [comment]
Thanks for your answer, which is fine with me for most of the use cases. For sure, I will use this feature!
Feel free to close the issue, or to keep it open, if you want to address my "cœur" replacement in a future release...

And many many thanks for Zim wiki :-)

**Comment by tildagail** [comment]
'œ' is in the Insert Symbol dialog. You can insert it with the shortcut \oelig by default and as a quick experiment I added an autocorrect using just oe. Just add this line somewhere to the symbols.list file and  then whenever you type oe it turns into œ:

`oe	339`

**Comment by gandrille** [comment]
@tildagail : Thanks to @jaap-karssenberg answer, I have already customized my symbols.list file in that direction. It's really effective for single symbol replacement :+1:

**Comment by introt** [comment]
>  Feel free to close the issue, or to keep it open, if you want to address my "cœur" replacement in a future release...

@gandrille Can this be achieved with the autocorrect plugin and a custom dictionary?

**Comment by gandrille** [comment]
@introt It is possible to have it replaced, but with a manual action every time the world is typed...which is not what I'm expecting.
 
At time being, using `symbols.list` file allows replacing few characters into a single one. For instance, --> is automatically replaced with a single character →
I would like a solution to replace few characters into few other characters.

I'm now using a walk around. I have a script for replacing some characters sequences with some others, using `sed` commands. Even if it requires the script to be launched manually, it is enough in order to fulfil my needs.

**Comment by introt** [comment]
Right, gotcha. Could you update the title (and the opening post) to something more descriptive, like "Text replacement (like symbols.list for words)"?

This could be achieved via a rather simple plugin; borrowing from [zim-auto-linker](https://github.com/HorseLuvver/zim-auto-linker/blob/main/autolink.py), the code would basically be the following + a GUI (the laborious part).

```
def on_end_of_word(self, textview, start, end, word, char, editmode):
	buffer = textview.get_buffer()
	if word in self.replacements.keys():
		with buffer.tmp_cursor(start):
			buffer.delete(start, end)
			buffer.insert_at_cursor(self.replacements[word])
		buffer.set_modified(True)
		textview.stop_emission('end_of_word')
```
